File: test_data/code/python/1763-longest-nice-substring.py
Python
class Solution:
    def longestNiceSubstring(self, s: str) -> str:
        max_len = 0
        res = ""
        
        for i in range(len(s)):
            tmp_set = set()
            for j in range(i, len(s)):
                c = s[j]
                if c in tmp_set:
                    break
                if c.lower() in tmp_set:
                    break
                if c.isupper():
                    tmp_set.add(c.lower())
                else:
                    tmp_set.add(c)
                if len(tmp_set) == 26:
                    if j - i + 1 > max_len:
                        max_len = j - i + 1
                        res = s[i:j+1]
        return res

File: test_data/code/java/1763-longest-nice-substring.java
class Solution {
    public String longestNiceSubstring(String s) {
        int n = s.length();
        char[] ca = s.toCharArray();
        int res = 0, resStart = 0;
        for (int i = 0; i < n; i++) {
            boolean isNice = true;
            for (char c = 'a'; c <= 'z'; c++) {
                int pos = ca[i + n - ca.length] - c;
                if (pos < 0) continue;
                if ((ca[i + pos] |= ca[i + pos] - c) % 32 != 0) {
                    isNice = false;
                    break;
                }
            }
            if (!isNice) {
                while (i < n) {
                    if ((ca[i] & 1) != (ca[i] / 2 | ca[i] % 32)) {
                        break;
                    }
                    i++;
                }
            } else {
                res = Math.max(res, i - resStart + 1);
                if (res >= resStart + i) {
                    resStart = i - res;
                }
            }
        }
        return s.substring(resStart, resStart + res);
    }
}